### DISCUSSION TITLE
How to access RTSP stream from inside a container

### DISCUSSION THREAD

**Discussion by datascientistlin** [bluenviron/mediamtx#1840] [Questions]
Dear All,

I have been trying to receive the RTSP stream published by rtsp-simple-server inside a container. The python script I'm running is as follows:

```
import cv2

rtsp_url = 'rtsp://<my-ip-address>:8554/stream1' 

cap = cv2.VideoCapture(rtsp_url)
while True:
    ret, frame = cap.read()
    if not ret:
        break
    # Process the frame or do something with it
    cv2.imshow('Frame', frame)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

cap.release()
```

I replaced the <my-ip-address> with my actual local IP address.

This script is working fine when I run it locally in my host machine. I could see the RTSP stream through `cv2.imshow`. However, when I run the same script from inside a container with opencv library, I got the following warnings. Nothing is streamed there.

```
[ WARN:0] global /opt/opencv-4.5.4/modules/videoio/src/cap_gstreamer.cpp (2060) handleMessage OpenCV | GStreamer warning: your GStreamer installation is missing a required plugin
[ WARN:0] global /opt/opencv-4.5.4/modules/videoio/src/cap_gstreamer.cpp (2075) handleMessage OpenCV | GStreamer warning: Embedded video playback halted; module source reported: Internal data stream error.
[ WARN:0] global /opt/opencv-4.5.4/modules/videoio/src/cap_gstreamer.cpp (1053) open OpenCV | GStreamer warning: unable to start pipeline
[ WARN:0] global /opt/opencv-4.5.4/modules/videoio/src/cap_gstreamer.cpp (616) isPipelinePlaying OpenCV | GStreamer warning: GStreamer: pipeline have not been created
```

Could you help me understand what adjustment I might need to take for accessing the RTSP stream from inside a container?

Thank you.

**Comment by aler9** [comment]
Hello, please post the server logs, that contain connection attempts to the server and can help you to identify the issue.

**Comment by datascientistlin** [comment]
this is resolved now

[a comment by an automated account is not shown]
